scripts/check-cxx-interop-compatibility.sh

```diff
@@ -30,6 +30,12 @@ package_name=$(swift package dump-package | jq -r '.name')
 
 cd "$working_dir"
 swift package init
+
+{
+  echo "let swiftSettings: [SwiftSetting] = [.interoperabilityMode(.Cxx)]"
+  echo "for target in package.targets { target.swiftSettings = swiftSettings }"
```

Comment:
This overwrites any existing build settings, could we do `target.swiftSettings.append(contentsOf: swiftSettings)` instead?

Reply:
I wondered about that approach myself and went back and forth. I landed on this approach because I thought it kept the test simple and targeted on the particular facet of the build we're interested in. Do you have specific reasons to use the combination?

Reply:
Yes, downgrading the Swift language mode to 5 is done this way, so the code above breaks that and builds it in Swift 6 mode (for packages with tools-version 6.0)